llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -16984,6 +16984,17 @@ SDValue RISCVTargetLowering::PerformDAGCombine(SDNode *N,
       return Op0.getOperand(0);
     }
 
+    if (ISD::isNormalLoad(Op0.getNode()) && Op0.hasOneUse() &&
```

Comment:
Is there something here preventing the load from having an offset?   I think the answer is that the load must be a vector load, and that we don't have base+offset addressing for that form, but could you add an assert here to confirm the offset is zero?

Reply:
I don't know what the offset field in LoadSDNode is for. We don't use it on RISC-V or X86. The immediate offset is represented by a separate ADD node in the base pointer that we pattern match at isel.